source/2_platform/rex_windows/src/gui_application.cpp

```diff
@@ -5,41 +5,186 @@
+      {
+        release_device_context(hwnd, s_glctx.dc);
+        REX_ERROR("Failed to set a compatible pixel format");
+        return false;
+      }
+
+      HGLRC temp = wglCreateContext(s_glctx.dc);
+      if (temp == NULL)
+      {
+        release_device_context(hwnd, s_glctx.dc);
+        REX_ERROR("Failed to create the initial rendering context");
+        return false;
+      }
+
+      wglMakeCurrent(s_glctx.dc, temp);
+
+      // Load WGL Extensions:
+      gladLoaderLoadWGL(s_glctx.dc);
+
+      s32 attribs[] =
+      {
+        WGL_CONTEXT_MAJOR_VERSION_ARB, 3,       // Set the MAJOR version of OpenGL to 3
+        WGL_CONTEXT_MINOR_VERSION_ARB, 2,       // Set the MINOR version of OpenGL to 2
+        WGL_CONTEXT_FLAGS_ARB, 
+        WGL_CONTEXT_FORWARD_COMPATIBLE_BIT_ARB, // Set our OpenGL context to be forward compatible
+        0
+      };
+
+      const char* extensions = (const char*)wglGetExtensionsStringARB(s_glctx.dc);
+      if (strstr(extensions, "WGL_ARB_create_context") == NULL)
+      {
+        s_glctx.glrc = wglCreateContextAttribsARB(s_glctx.dc, NULL, attribs);
+        if (s_glctx.glrc == NULL)
+        {
+          wglDeleteContext(temp);
+          release_device_context(hwnd, s_glctx.dc);
+          REX_ERROR("Failed to create Windows OpenGL Context");
+
+          return false;
+        }
+
+        wglMakeCurrent(NULL, NULL);
+        wglDeleteContext(temp);
+        wglMakeCurrent(s_glctx.dc, s_glctx.glrc);
+      }
+      else
+      {
+        s_glctx.glrc = temp;
+      }
+
+      if (!gladLoaderLoadGL()) 
+      {
+        wglMakeCurrent(NULL, NULL);
+        wglDeleteContext(s_glctx.glrc);
+        release_device_context(hwnd, s_glctx.dc);
+        REX_ERROR("Glad Loader failed!");
+        
+        return false;
+      }
+
+      return true;
+    }
+  }
+}
+
+#define create_ctx(hwnd) rex::opengl::create_gl_context(hwnd)
+#else
+#define create_ctx(hwnd) (hwnd == nullptr)
+#endif
+
 namespace rex
 {
   namespace win32
   {
     struct GuiApplication::Internal
     {
       Internal(const PlatformCreationParams& guiCreationParams, const ApplicationCreationParams& appParams, CommandLineArguments&& genericCreationParams)
-          : platform_creation_params(rsl::move(guiCreationParams))
-          , app_params(appParams)
-          , cmd_line_args(rsl::move(genericCreationParams))
+        : platform_creation_params(rsl::move(guiCreationParams))
+        , app_params(appParams)
+        , cmd_line_args(rsl::move(genericCreationParams))
       {
       }
 
       bool initialize()
       {
         window = create_window();
-        if(window == nullptr)
+        if (window == nullptr)
+        {
+          return false;
+        }
+
+        HWND hwnd = (HWND)window->get_primary_display_handle();
+        if (create_ctx(hwnd) == false)
+        {
+          return false;
+        }
+
+        if (renderer::initialize((void*)&hwnd, app_params.max_render_commands) == false)
```

Comment:
I think for now this is fine, but in the future I'd like to get rid of the void* cast. But we can wait for doing that when we know what the userData will be used for

Reply:
Yeah, I was not happy with the design here but currently, I did not see an alternative on how to pass the window handle in a cross-platform way. So I did not give it much thought. 

This will change in the future once other platforms are in place.